filer/admin/fileadmin.py

```diff
@@ -64,31 +72,34 @@ def response_change(self, request, obj):
         Overrides the default to be able to forward to the directory listing
         instead of the default change_list_view
         """
-        r = super(FileAdmin, self).response_change(request, obj)
-        if 'Location' in r and r['Location']:
-            # it was a successful save
-            if (r['Location'] in ['../'] or
-                    r['Location'] == self._get_post_url(obj)):
-                # this means it was a save: redirect to the directory view
-                if obj.folder:
-                    url = reverse('admin:filer-directory_listing',
-                                  kwargs={'folder_id': obj.folder.id})
-                else:
-                    url = reverse(
-                        'admin:filer-directory_listing-unfiled_images')
-                url = "%s%s%s" % (url, popup_param(request),
-                                  selectfolder_param(request, "&"))
-                return HttpResponseRedirect(url)
+        admin_url_params = AdminUrlParams(request)
+        if (
+            request.POST and
+            admin_url_params.popup and
+            admin_url_params.pick and
+            '_continue' not in request.POST
+        ):
+            # popup in pick mode. response_change only gets called if the
+            # operation was successful. In pick mode we just want to go
+            # back to the folder list view and keep the params.
+            if obj.folder:
+                url = reverse('admin:filer-directory_listing',
+                              kwargs={'folder_id': obj.folder.id})
             else:
-                # this means it probably was a save_and_continue_editing
-                pass
-        return r
+                url = reverse(
+                    'admin:filer-directory_listing-unfiled_images')
+            url = "{}{}".format(
```

Comment:
AFAIK it is only python<=2.6 that don't support that format. We're only supporting python 2.7+